Error Title: NullPointerException in WhitelistCommand Execution

🐞 Bug Feedback | BUG反馈

📝 Bug Description | BUG描述
A NullPointerException occurred during the execution of the WhitelistCommand at line 93. The issue is within the OfflinePlayer class, specifically in the setWhitelisted method at line 150. The error is triggered when attempting to invoke the toLowerCase method on the return value of OfflinePlayer.getName(), which is found to be null. This error is further propagated through the command execution flow.

🔍 Possible Cause | 可能的原因
The NullPointerException arises when trying to convert the player's name to lowercase in order to whitelist them. The error suggests that the getName() method of OfflinePlayer is returning null. This could happen if the OfflinePlayer instance represents a player with no assigned name or if there is an issue fetching the player's name.

💻 OS and versions | 操作系统和核心版本
Nukkit-MOT master/8cac79a
Version: 1.20.40 (protocol 622)
Java: OracleJDK
Java version: 17
Platform: win10

```error
20:38:28 [ERROR] Throwing
java.lang.NullPointerException: Cannot invoke "String.toLowerCase()" because the return value of "cn.nukkit.OfflinePlayer.getName()" is null
at cn.nukkit.OfflinePlayer.setWhitelisted(OfflinePlayer.java:150) ~[Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.command.defaults.WhitelistCommand.execute(WhitelistCommand.java:93) ~[Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:262) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Server.dispatchCommand(Server.java:959) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Player.handleDataPacket(Player.java:3884) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.network.session.RakNetPlayerSession.serverTick(RakNetPlayerSession.java:199) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.network.RakNetInterface.process(RakNetInterface.java:95) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.network.Network.processInterfaces(Network.java:119) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Server.tick(Server.java:1402) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Server.tickProcessor(Server.java:1136) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Server.start(Server.java:1104) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Server.<init>(Server.java:818) [Nukkit-MOT-SNAPSHOT.jar:?]
at cn.nukkit.Nukkit.main(Nukkit.java:87) [Nukkit-MOT-SNAPSHOT.jar:?] 
```

--- a/src/main/java/cn/nukkit/Server.java
+++ b/src/main/java/cn/nukkit/Server.java
@@ -1798,7 +1798,7 @@ public class Server {
             return result;
         }
 
-        return lookupName(name).map(uuid -> new OfflinePlayer(this, uuid))
+        return lookupName(name).map(uuid -> new OfflinePlayer(this, uuid, name))
                 .orElse(new OfflinePlayer(this, name));
     }
 

